marklogic-client-api/src/test/java/com/marklogic/client/test/document/DocumentWriteOperationTest.java

```diff
@@ -309,6 +308,56 @@ public void DocumentWriteSetWithDisableMetadataTest_2() {
         textDocumentManager.write(batch);
     }
 
+    @Test
+    public void DocumentQueryWithForest() {
+        DocumentWriteSet batch = textDocumentManager.newWriteSet();
+        batch.addDefault(defaultMetadata1);
+
+        for (int i = 0; i < 30; i++) {
+            String uri = "doc" + i + ".txt";
+            StringHandle handle = new StringHandle("Document - " + i).withFormat(Format.TEXT);
+            batch.add(uri, handle);
+        }
+
+        textDocumentManager.write(batch);
+        StructuredQueryDefinition query = new StructuredQueryBuilder().collection(collectionName);
+
+        DocumentPage documentsF1 = textDocumentManager.search(query, 1, "java-unittest-1");
+        DocumentPage documentsF2 = textDocumentManager.search(query, 1, "java-unittest-2");
+        DocumentPage documentsF3 = textDocumentManager.search(query, 1, "java-unittest-3");
```

Comment:
It might be slightly easier to maintain if the test 

- declared a count on the number of forests
- instantiated a DocumentPage array with the count as length
- iterated from 0 to i < count to initialize each array item with a search result, sum the results, and construct sets

Given that it's a test, however, the change isn't mandatory.

Reply:
Yes, that's cleaner! Changed to below code:
```
int forestCount = 3;
        DocumentPage[] documents = new DocumentPage[forestCount];
        ArrayList<Set<String>> sets = new ArrayList<Set<String>>();

        long totalCount = 0;

        for (int i = 0; i < forestCount; i++) {
            documents[i] = textDocumentManager.search(query, 1, "java-unittest-" + String.valueOf(i+1));
            totalCount += documents[i].getTotalSize();
            sets.add(new HashSet<String>());
            for (DocumentRecord document : documents[i]) {
                sets.get(i).add(document.getUri());
            }

        }
        assertEquals(30, totalCount);
```